stdlib/buffer.ml

```diff
@@ -87,102 +96,91 @@ let resize b more =
   let new_buffer = Bytes.create !new_len in
   (* PR#6148: let's keep using [blit] rather than [unsafe_blit] in
      this tricky function that is slow anyway. *)
-  Bytes.blit b.buffer 0 new_buffer 0 b.position;
-  b.buffer <- new_buffer;
-  b.length <- !new_len;
-  assert (b.position + more <= b.length);
-  assert (old_pos + more <= b.length);
+  Bytes.blit b.inner.buffer 0 new_buffer 0 b.position;
+  b.inner <- { buffer = new_buffer; length = !new_len };
+  assert (b.position + more <= b.inner.length);
+  assert (old_pos + more <= b.inner.length);
```

Comment:
I would get rid of the assert `b.position + more <= b.inner.length`. It was previously used to justify using unsafe writes in the `add_*` functions, but we cannot use it anymore as it may be now violated by races at any time. In fact I would just get rid of the two asserts.

Reply:
Removing these two asserts that may fail randomly (due to a race) sounds like a good idea to me too (since `assert` are not good error messages).